File: app/src/main/java/de/garten/training/depotflow/domain/WorkOrderStatus.java
package de.garten.training.depotflow.domain;

public enum WorkOrderStatus {
    NEW("new"),
    ACCEPTED("accepted"),
    IN_PROGRESS("in_progress"),
    BLOCKED("blocked"),
    DONE("done"),
    CANCELLED("cancelled");

    private final String serverValue;

    WorkOrderStatus(String serverValue) {
        this.serverValue = serverValue;
    }

    public String toServerValue() {
        return serverValue;
    }

    public static WorkOrderStatus fromServerValue(String value) {
        if (value == null) {
            return NEW;
        }
        for (WorkOrderStatus status : values()) {
            if (status.serverValue.equals(value)) {
                return status;
            }
        }
        return NEW;
    }

    public boolean isTerminal() {
        return this == DONE || this == CANCELLED;
    }
}

File: app/src/main/java/de/garten/training/depotflow/domain/WorkOrderStatus.kt
package de.garten.training.depotflow.domain

enum class WorkOrderStatus(private val serverValue: String) {
    NEW("new"),
    ACCEPTED("accepted"),
    IN_PROGRESS("in_progress"),
    BLOCKED("blocked"),
    DONE("done"),
    CANCELLED("cancelled");

    fun toServerValue(): String = serverValue

    fun isTerminal(): Boolean = this == DONE || this == CANCELLED

    companion object {
        @JvmStatic
        fun fromServerValue(value: String?): WorkOrderStatus {
            return entries.firstOrNull { it.serverValue == value } ?: NEW
        }
    }
}
